includes/Core/Modules/Modules.php

```diff
@@ -428,6 +430,31 @@ function( Module $module ) {
 		return $data;
 	}
 
+	/**
+	 * Populates modules data to pass to JS.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $modules_data Inline modules data.
+	 * @return array Inline modules data.
+	 */
+	private function inline_modules_data( $modules_data ) {
+		$available_modules = $this->get_available_modules();
+
+		$modules_with_data_available_state = array_filter(
+			$available_modules,
+			function( Module $module ) {
+				return $module instanceof Module_With_Data_Available_State;
+			}
+		);
+
+		foreach ( $modules_with_data_available_state as $module ) {
+			$modules_data[ 'data_available_' . $module->slug ] = $module->is_connected() && $module->is_data_available();
```

Comment:
Let's also check that the module is active (not only connected) because the module's method will not check this. It would be difficult/unlikely for this to happen I think but worth adding for safety. On the client side, `isModuleConnected` checks this internally, for some reason we don't do the same in the `Modules::is_module_connected` method 😄